scripts/python-skywater-pdk/skywater_pdk/fet_simulator/fet_simulator.py

```diff
@@ -0,0 +1,293 @@
+    iparam:
+        template string for creating internal parameter names (gm, id, gds,
+        cgg) for a given FET
+    fet_W: FET width
+    """
+    sim = c.simulator()
+    sim.save_internal_parameters(
+        iparam % 'gm', iparam % 'id', iparam % 'gds', iparam % 'cgg'
+    )
+
+    try:
+        # run the dc simulation
+        an = sim.dc(Vgg=slice(0, 1.8, 0.01))
+
+        # calculate needed values..need as_ndarray() since most of these have
+        # None as the unit and that causes an error
+        gm_id = (an.internal_parameters[iparam % 'gm'].as_ndarray() /
+                 an.internal_parameters[iparam % 'id'].as_ndarray())
+        ft = (an.internal_parameters[iparam % 'gm'].as_ndarray() /
+              an.internal_parameters[iparam % 'cgg'].as_ndarray())
+        id_W = (an.internal_parameters[iparam % 'id'].as_ndarray() / fet_W)
+        gm_gds = (an.internal_parameters[iparam % 'gm'].as_ndarray() /
+                  an.internal_parameters[iparam % 'gds'].as_ndarray())
+    except Exception:
+        sim.ngspice.destroy('all')
+        sim.ngspice.reset()
+        raise
+
+    sim.ngspice.destroy('all')
+    sim.ngspice.reset()
+
+    return gm_id, ft, id_W, gm_gds, an.nodes['v-sweep']
+
+
+def init_plots(fet_name, W):
+    """
+    Initializes plots for FET bins simulation.
+
+    Parameters
+    ----------
+    fet_name: name of the current FET cell
+    W: FET width
+    """
+    figs = [plt.figure(), plt.figure(), plt.figure(), plt.figure()]
+    plts = [f.subplots() for f in figs]
+    figs[0].suptitle(f'{fet_name} Id/W vs gm/Id (W = {W})')
+    plts[0].set_xlabel("gm/Id")
+    plts[0].set_ylabel("Id/W")
+    plts[0].grid(True)
+    figs[1].suptitle(f'{fet_name} fT vs gm/Id (W = {W})')
+    plts[1].set_xlabel("gm/Id")
+    plts[1].set_ylabel("f_T")
+    plts[1].grid(True)
+    figs[2].suptitle(f'{fet_name} gm/gds vs gm/Id (W = {W})')
+    plts[2].set_xlabel("gm/Id")
+    plts[2].set_ylabel("gm/gds")
+    plts[2].grid(True)
+    figs[3].suptitle(f'{fet_name} gm/Id vs Vgg (W = {W})')
+    plts[3].set_xlabel("Vgg")
+    plts[3].set_ylabel("gm/Id")
+    plts[3].grid(True)
+    return figs, plts
+
+
+def gen_plots(gm_id, id_W, ft, gm_gds, vsweep, fet_W, fet_L, plts):
+    """
+    Generates plot lines for FET bins simulation parameters.
+
+    Parameters
+    ----------
+    gm_id: gm/Id values
+    id_W: Id/W values
+    ft: f_T values
+    gm_gds: gm/gds values
+    vsweep: v-sweep values
+    fet_W: FET width
+    fet_L: FET length
+    plts: plots on which plot lines should be drawn
+    """
+    # plot some interesting things
+    plts[0].plot(gm_id, id_W, label=f'W {fet_W} x L {fet_L}')
+    plts[1].plot(gm_id, ft, label=f'W {fet_W} x L {fet_L}')
+    plts[2].plot(gm_id, gm_gds, label=f'W {fet_W} x L {fet_L}')
+    plts[3].plot(vsweep, gm_id, label=f'W {fet_W} x L {fet_L}')
+
+
+def close_plots(figs):
+    """
+    Closes plots.
+    """
+    for f in figs:
+        plt.close(f)
+
+
+def read_bins(fname):
```

Comment:
Reading bins is something you want to do for multiple reasons (not just simulation). This should probably be part of a generic the `cell` class @ https://github.com/google/skywater-pdk/blob/master/scripts/python-skywater-pdk/skywater_pdk/base.py#L510 ?

Reply:
Moved the method to Cell class